documentation/assets/Example: training an external model.md

```diff
@@ -0,0 +1,753 @@
+  (decoder3): Sequential(
+    (dec3conv1): Conv2d(256, 128, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
+    (dec3norm1): BatchNorm2d(128, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
+    (dec3relu1): ReLU(inplace=True)
+    (dec3conv2): Conv2d(128, 128, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
+    (dec3norm2): BatchNorm2d(128, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
+    (dec3relu2): ReLU(inplace=True)
+  )
+  (upconv2): ConvTranspose2d(128, 64, kernel_size=(2, 2), stride=(2, 2))
+  (decoder2): Sequential(
+    (dec2conv1): Conv2d(128, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
+    (dec2norm1): BatchNorm2d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
+    (dec2relu1): ReLU(inplace=True)
+    (dec2conv2): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
+    (dec2norm2): BatchNorm2d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
+    (dec2relu2): ReLU(inplace=True)
+  )
+  (upconv1): ConvTranspose2d(64, 32, kernel_size=(2, 2), stride=(2, 2))
+  (decoder1): Sequential(
+    (dec1conv1): Conv2d(64, 32, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
+    (dec1norm1): BatchNorm2d(32, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
+    (dec1relu1): ReLU(inplace=True)
+    (dec1conv2): Conv2d(32, 32, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
+    (dec1norm2): BatchNorm2d(32, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
+    (dec1relu2): ReLU(inplace=True)
+  )
+  (conv): Conv2d(32, 1, kernel_size=(1, 1), stride=(1, 1))
+)
+```
+
+## 3. Loss function
+
+While a popular choice for binary segmentation loss function is the binary cross-entropy (BCE) loss, often it is
+the intersection-over-union (IoU), or the [Jaccard Index](https://en.wikipedia.org/wiki/Jaccard_index), that serves
+as a measure of success. In this example, we will use a combination of the BCE and IoU as the loss function. While
+SuperGradients provides, among many other losses, an implementation of the combined BCE and Dice loss, which could be 
+used for our purposes as well, we will show how we can define our own user-defined loss function and train our model 
+with it using SuperGradients.
+
+Similar to using an external model, the custom loss function's class must inherit from `torch.nn.Module`. The 
+`forward()` function's first parameter needs to be the predictions tensor and the second parameter needs to be the
+targets tensor.
+
+```
+import torch
+import torch.nn as nn
+from torchmetrics import JaccardIndex
+
+
+class CustomSegLoss(torch.nn.Module):
+    def __init__(self, bce_weight=1, iou_weight=1):
+        super(CustomSegLoss, self).__init__()
+
+        self.bce_weight = bce_weight
+        self.iou_weight = iou_weight
+
+        self.bce_loss = nn.BCELoss()
+        self.iou_func = JaccardIndex(num_classes=2)
```

Comment:
from torchmetrics.classification import JaccardIndex
JaccardIndex.is_differentiable
Out[3]: False

Its a good idea to add iou component to the loss, but its not what is happening here. You are basically training just on BCE + some numbers as far as I can see.

Reply:
Implemented a custom IoU loss, which is differentiable